assets/src/blocks/components/product-attributes/index.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Product attributes component.
+ */
+
+
+import { getProductAttributes } from '../utilities';
+import FastWCProductAttribute from './attribute'
+
+import PropTypes from 'prop-types';
+
+const { __ } = wp.i18n;
+const { useState, useEffect } = wp.element;
+
+const FastWCProductAttributes = ( {
+	onChange,
+	product,
+	variant,
+	selected,
+} ) => {
+	const [attKeys, setAttKeys] = useState([]);
+	const [labels, setLabels] = useState([]);
+	const [options, setOptions] = useState([]);
+
+	useEffect(
+		() => {
+			getProductAttributes( product, variant )
+				.then( ( list ) => {
+					const newOptions = {};
+					const newAttributes = {};
+
+					Object.keys( selected ).map( ( selectedKey ) => {
+						let availableValues = [];
+
+						if ( list.options && list.options[ selectedKey ] && list.options[ selectedKey ].length ) {
+							availableValues = list.options[ selectedKey ].map( ( option ) => option.value );
+						}
+
+						if ( list.attKeys && list.attKeys.includes( selectedKey ) && availableValues.includes( selected[ selectedKey ] ) ) {
+							newAttributes[ selectedKey ] = selected[ selectedKey ];
+						}
+					} );
+
+					if ( list.attKeys && list.attKeys.length ) {
+						list.attKeys.map( ( attKey ) => {
+							if ( list.values[ attKey ] ) {
+								newAttributes[ attKey ] = list.values[ attKey ];
+								newOptions[ attKey ] = [
+									{
+										value: list.values[ attKey ],
+										label: list.values[ attKey ],
+									},
+								];
+							} else {
+								newOptions[ attKey ] = list.options[ attKey ];
+								if ( ! newAttributes[ attKey ] && list.options[ attKey ].length ) {
+									const option = list.options[ attKey ][0];
+									newAttributes[ attKey ] = option.value;
+								}
+							}
+						} );
+
+
+
+						setAttKeys( list.attKeys );
+						setLabels( list.labels );
+					}
+
+					onChange( newAttributes );
+					setOptions( newOptions );
+				} );
+		},
+		[
+			product,
+			variant,
+		]
+	);
+
+	const onAttributeChange = ( attKey, value ) => {
```

Comment:
nit: to match our react style: we have all props for these named `onX` and the actual functions be called `handleX` to disambiguate which is a prop and which is the actual handler
```suggestion
	const handleAttributeChange = ( attKey, value ) => {
```